### PR TITLE
Implement mediaMTX RTMP auth

### PR DESCRIPTION
**Pull request by mjh1**
For the mediaMTX config I enabled the control api with `api: yes` and updated runOnReady: `runOnReady: curl http://localhost:5936/live/video-to-video/start -F stream=$MTX_PATH -F source_id=$MTX_SOURCE_ID`

**What does this pull request do? Explain your changes. (required)**

Adding auth via a webhook call for the rtmp input stream.

We do also want to retrieve some config params as the response to the auth webhook but afaik those aren't defined yet and I figured it's probably nicer in a separate PR anyway. @j0sh if you'd rather I do it in this PR let me know or go a bit further with adding in placeholders.

**Specific updates (required)**

- 
- 
- 

**How did you test each of these updates (required)**

Tested locally running mediaMTX and the gateway manually

**Does this pull request close any open issues?**

**Checklist:**

- [x] Read the [contribution guide](./CONTRIBUTING.md)
- [x] `make` runs successfully
- [x] All tests in `./test.sh` pass
- [ ] README and other documentation updated
- [ ] [Pending changelog](./CHANGELOG_PENDING.md) updated

### CODE REVIEW

File: media/rtmp2segment.go [hunk 1]
```diff
@@ -219,6 +223,7 @@ func processSegments(segmentHandler SegmentHandler, outFilePattern string, compl
 }
 
 func readSegment(segmentHandler SegmentHandler, file *os.File, pipeName string) {
+	log.Println("readSegment")
```
Comment (j0sh) [inline media/rtmp2segment.go]: stray? or at least `slog.Debug` ? (I think this is the only use of the `log` import in this file)

File: server/ai_mediaserver.go [hunk 2]
```diff
@@ -360,10 +360,30 @@ func (ls *LivepeerServer) StartLiveVideo() http.Handler {
...
 		if streamName == "out-stream" {
 			// skip for now; we don't want to re-publish our own outputs
 			return
 		}
+
+		err := authenticateAIStream(AuthWebhookURL, AIAuthRequest{
+			Stream: streamName,
+		})
+		if err != nil {
+			kickErr := kickRTMPConnection(sourceID)
```
Comment (j0sh) [inline server/ai_mediaserver.go]: If the source is WHIP will this still work?

File: media/rtmp2segment.go [hunk 3]
```diff
@@ -36,15 +36,19 @@ func (ms *MediaSegmenter) RunSegmentation(in string, segmentHandler SegmentHandl
...
+	_, err := ffmpeg.Transcode3(&ffmpeg.TranscodeOptionsIn{
 		Fname: in,
 	}, []ffmpeg.TranscodeOptions{{
 		Oname:        outFilePattern,
 		AudioEncoder: ffmpeg.ComponentOptions{Name: "copy"},
 		VideoEncoder: ffmpeg.ComponentOptions{Name: "copy"},
 		Muxer:        ffmpeg.ComponentOptions{Name: "segment"},
 	}})
+	if err != nil {
+		slog.Error("Failed to run segmentation:", "err", err)
```
Comment (j0sh) [inline media/rtmp2segment.go]: Might be worth including the input here so it is easier to track which job it belongs to

```suggestion
		slog.Error("Failed to run segmentation", "in", in, "err", err)
```

File: server/auth.go [hunk 4]
```diff
@@ -95,3 +95,39 @@ func (a authWebhookResponse) areProfilesEqual(b authWebhookResponse) bool {
 
 	return string(profilesA) == string(profilesB)
 }
+
+type AIAuthRequest struct {
+	Stream string `json:"stream"`
+	// TODO not sure what params we need yet
```
Comment (j0sh) [inline server/auth.go:101]: later on, query params

**Review by j0sh — APPROVED** [review]
Looks good, just double checking it also works to terminate a WHIP stream.

File: media/rtmp2segment.go (hunk 1 above)
Comment (mjh1) [reply media/rtmp2segment.go]: thanks yeah meant to remove 👍

File: server/ai_mediaserver.go (hunk 2 above)
Comment (mjh1) [reply server/ai_mediaserver.go]: ah 🤦 stupidly forgot, fixed that now 👍

**Pull request merged**